WalletWasabi/WabiSabi/Client/HttpRequestHandlerProxy.cs

```diff
@@ -0,0 +1,74 @@
+using Newtonsoft.Json;
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using WalletWasabi.Tor.Http;
+using WalletWasabi.Tor.Http.Extensions;
+using WalletWasabi.WabiSabi.Backend.PostRequests;
+using WalletWasabi.WabiSabi.Models;
+
+namespace WalletWasabi.WabiSabi.Client
+{
+	public class HttpRequestHandlerProxy : IArenaRequestHandler
+	{
+		private IHttpClient _client;
+
+		public HttpRequestHandlerProxy(IHttpClient client)
+		{
+			_client = client;
+		}
+
+		public Task<InputRegistrationResponse> RegisterInputAsync(InputRegistrationRequest request) =>
+			SendAndReceiveAsync<InputRegistrationRequest, InputRegistrationResponse>("register-input", request);
+
+		public Task<ConnectionConfirmationResponse> ConfirmConnectionAsync(ConnectionConfirmationRequest request) =>
+			SendAndReceiveAsync<ConnectionConfirmationRequest, ConnectionConfirmationResponse>("confirm-connection", request);
+
+		public Task<OutputRegistrationResponse> RegisterOutputAsync(OutputRegistrationRequest request) =>
+			SendAndReceiveAsync<OutputRegistrationRequest, OutputRegistrationResponse>("register-output", request);
+
+		public Task<ReissueCredentialResponse> ReissueCredentialAsync(ReissueCredentialRequest request) =>
+			SendAndReceiveAsync<ReissueCredentialRequest, ReissueCredentialResponse>("reissue-credential", request);
```

Comment:
I think I would prefer if the action was an enum instead of a string

Reply:
Yes. You're right.

Reply:
Done.